Goal: set PATH according to ENV_SUPATH and ENV_PATH (for login), as for
      su.
Fixes: #330803

Status wrt upstream:

Index: shadow-4.0.15/libmisc/setupenv.c
===================================================================
--- shadow-4.0.15.orig/libmisc/setupenv.c	2006-03-08 19:32:10.692126935 +0100
+++ shadow-4.0.15/libmisc/setupenv.c	2006-03-08 19:33:40.256983644 +0100
@@ -242,7 +242,6 @@
 	addenv ("USER", info->pw_name);
 	addenv ("LOGNAME", info->pw_name);
 
-#ifndef USE_PAM
 	/*
 	 * Create the PATH environmental variable and export it.
 	 */
@@ -259,6 +258,7 @@
 		/* only value specified without "PATH=" */
 		addenv ("PATH", cp);
 	}
+#ifndef USE_PAM
 
 	/*
 	 * Create the MAIL environmental variable and export it.  login.defs
